### PR TITLE
Allow for returning DependencyCollector from DSL methods

### PR DESCRIPTION
**Pull request by tresat**
Having this type annotated would be helpful for our declarative experiments on the JVM team.  Could we add this to your development branch?

Update: Looks like we can merge this into master now.  
Update: Maybe not?

### CODE REVIEW

File: subprojects/core-api/src/main/java/org/gradle/api/artifacts/dsl/DependencyCollector.java [hunk 1]
```diff
@@ -16,6 +16,7 @@
 
 package org.gradle.api.artifacts.dsl;
 
+import com.h0tk3y.kotlin.staticObjectNotation.Restricted;
```
Comment (jvandort) [inline subprojects/core-api/src/main/java/org/gradle/api/artifacts/dsl/DependencyCollector.java:19]: Why is this not a Gradle package name? 

I think we should update the prototype to use a Gradle package before we start using it in the gradle codebase

Comment (tresat) [reply subprojects/core-api/src/main/java/org/gradle/api/artifacts/dsl/DependencyCollector.java:19]: Good catch - I didn't notice the packages had changed after the merge to `master`.  I'll update this to use the new `org.gradle` ones.

Comment (tresat) [reply subprojects/core-api/src/main/java/org/gradle/api/artifacts/dsl/DependencyCollector.java:19]: Hmmm, this isn't true.  The Restricted annotation hasn't moved to `org.gradle.internal.restricteddsl`.

Comment (jvandort) [reply subprojects/core-api/src/main/java/org/gradle/api/artifacts/dsl/DependencyCollector.java:19]: https://github.com/gradle/gradle/blob/74b5468e20a009b1bdc532fd5c88bb94248d8a6b/platforms/core-configuration/declarative-dsl-api/src/main/java/org/gradle/declarative/dsl/model/annotations/Restricted.java#L23

Not that one?

**Pull request closed without merging**